cmd/noms/noms_commit.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/attic-labs/noms/cmd/util"
+	"github.com/attic-labs/noms/go/d"
+	"github.com/attic-labs/noms/go/dataset"
+	"github.com/attic-labs/noms/go/spec"
+	"github.com/attic-labs/noms/go/types"
+	flag "github.com/juju/gnuflag"
+)
+
+const (
+	dateFormat = "2006-01-02T15:04:05-0700"
+)
+
+var (
+	allowDupe   bool
+	date        string
+	message     string
+	metaStrings string
+	metaPaths   string
+)
+
+var nomsCommit = &util.Command{
+	Run:       runCommit,
+	UsageLine: "commit [options] <dataset> [absolute-path]",
+	Short:     "Commits a specified value as head of the dataset",
+	Long:      "If absolute-path is not provided, then it is read from stdin. See Spelling Objects at https://github.com/attic-labs/noms/blob/master/doc/spelling.md for details on the dataset and absolute-path arguments.",
+	Flags:     setupCommitFlags,
+	Nargs:     1, // if absolute-path not present we read it from stdin
+}
+
+func setupCommitFlags() *flag.FlagSet {
+	commitFlagSet := flag.NewFlagSet("commit", flag.ExitOnError)
+	commitFlagSet.BoolVar(&allowDupe, "allow-dupe", false, "creates a new commit, even if it would be identical (modulo metadata and parents) to the existing HEAD.")
+	commitFlagSet.StringVar(&date, "date", "", "alias for -meta 'date:<date>'. '<date>' must be iso8601-formatted. If '<date>' is empty, it defaults to the current date.")
+	commitFlagSet.StringVar(&message, "message", "", "alias for -meta 'message:<message>'")
+	commitFlagSet.StringVar(&metaStrings, "meta", "", "'<key>:<value>' - creates a metadata field called 'key' set to 'value'. Value should be human-readable encoded.")
+	commitFlagSet.StringVar(&metaPaths, "meta-p", "", "'<key>:<path>' - creates a metadata field called 'key' set to the value at <path>")
+	return commitFlagSet
+}
+
+func runCommit(args []string) int {
+	ds, err := spec.GetDataset(args[0])
+	d.CheckError(err)
+	defer ds.Database().Close()
+
+	var path string
+	if len(args) == 2 {
+		path = args[1]
+	} else {
+		readPath := make([]byte, 256)
+		n, err := os.Stdin.Read(readPath)
```

Comment:
I don't believe that this API is guaranteed to return you a line. You might get less or more than a line. You should definitely use ReadLine().